src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/PrecompiledViewInfo.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Mvc.Core.ApplicationParts
+{
+    public class PrecompiledViewInfo
```

Comment:
Just a note, not sure if there's anything that can be done but this is darn close to the `CompileOutput` API. Only difference is once has the assembly factory and one has the assembly.